core/src/apps/ethereum/sign_tx.py

```diff
@@ -22,8 +25,27 @@
 MAX_CHAIN_ID = 2147483629
 
 
+def format_access_list(access_list):
+    mapped = map(
+        lambda item: [
+            address.bytes_from_address(item.address),
+            list(map(lambda s: unhexlify(s[2:]), item.storage_keys)),
+        ],
+        access_list,
+    )
+    return list(mapped)
+
+
 @with_keychain_from_chain_id
 async def sign_tx(ctx, msg, keychain):
+
+    if msg.tx_type == 2:
```

Comment:
does it make sense to set `msg.tx_type == 2` and at the same time `msg.eip2718` is False? what happens then?

Reply:
No, it doesn't. My latest commit adds a check for that.

Reply:
Feel free to let me know if you would prefer something else.

Reply:
The `check` function will throw if the `eip2718_type` is out of range. So I think my latest commit should be fine for this.

Reply:
But let me know if that isn't the case.